tests/rptest/tests/e2e_iam_role_test.py

```diff
@@ -0,0 +1,206 @@
+from ducktape.mark.resource import cluster
+
+from rptest.clients.types import TopicSpec
+from rptest.services.mock_iam_roles_server import MockIamRolesServer
+from rptest.tests.e2e_shadow_indexing_test import EndToEndShadowIndexingBase
+from rptest.util import produce_until_segments, wait_for_segments_removal
+
+
+class AWSRoleFetchTests(EndToEndShadowIndexingBase):
+    def __init__(self, test_context, extra_rp_conf=None):
+        self.iam_server = MockIamRolesServer(test_context,
+                                             'aws_iam_role_mock.py')
+        if not extra_rp_conf:
+            extra_rp_conf = {}
+
+        extra_rp_conf[
+            'cloud_storage_credentials_source'] = 'aws_instance_metadata'
+        super().__init__(test_context,
+                         extra_rp_conf,
+                         environment={
+                             'RP_SI_CREDS_API_HOST': self.iam_server.hostname,
+                             'RP_SI_CREDS_API_PORT': self.iam_server.port,
+                         })
+
+    def setUp(self):
+        self.iam_server.start()
+        super().setUp()
+
+    def tearDown(self):
+        super().tearDown()
+        self.iam_server.stop()
+        self.iam_server.wait()
+        self.iam_server.clean()
+
+    @cluster(num_nodes=6)
```

Comment:
You probably want to add `log_allow_list` here and in other test cases.

Reply:
added